pkg/apiserver/watch.go

```diff
@@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				// client disconnect.
 				return
 			}
-			flusher.Flush()
+			if len(ch) == 0 {
```

Comment:
also shouldnt http layer will do a auto flush if the http buffer is getting large?